dbt/macros/no_null_cols.sql

```diff
@@ -1,19 +1,23 @@
-{% test no_null_cols(model, filter_condition=None, filtered_columns=[]) %}
+{% test no_null_cols(model, partial_columns=[]) %}
 
 WITH column_checks AS (
-    {% set columns = dbt_utils.get_columns_in_relation(model) %}
+    {% set columns = adapter.get_columns_in_relation(model) %}
 
     {% for column in columns %}
         {% set column_name = column.name %}
+        {% set condition_sql = "1=1" %}  -- Default: test all rows
+
+        {# Check if this column has a specific condition #}
+        {% for partial_col in partial_columns %}
+            {% if partial_col.get('column_name') == column_name %}
+                {% set condition_sql = partial_col.get('expect_at_least_one_value_when') %}
+            {% endif %}
+        {% endfor %}
```

Comment:
Just noting this is O(mn). Shouldn't be a big deal for currently expected values of m and n, but if we start seeing performance problems we can switch to a map implementation for `partial_columns`, like

```suggestion
        {% if column_name in partial_columns %}
            {% set condition_sql = partial_columns.get(column_name) %}
        {% endif %}
```
we'd lose the nice `expect_at_least_one_value_when` syntax cue though, so probably something to save for later